packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -2224,6 +2224,28 @@ class IterableProperty<T> extends DiagnosticsProperty<Iterable<T>> {
     if (value.isEmpty)
       return ifEmpty ?? '[]';
 
+    // Use debugFormatDouble to format Iterable<double>.
+    if (T == double) {
+      final StringBuffer sb = StringBuffer();
+      if (parentConfiguration != null && !parentConfiguration.lineBreakProperties) {
+        for (dynamic item in value) {
+          if (sb.isNotEmpty) {
+            sb.write(', ');
+          }
+          sb.write(debugFormatDouble(item));
+        }
+        return '[${sb.toString()}]';
+      } else {
+        final bool isSingleLine = _isSingleLine(style);
+        for (dynamic item in value) {
+          if (sb.isNotEmpty)
+            sb.write(isSingleLine ? ', ' : '\n');
+          sb.write(debugFormatDouble(item));
+        }
+        return sb.toString();
+      }
+    }
+
```

Comment:
Why not support this without forking the cases for regular and double properties?
If you wrote
```dart
Iterable<String> formattedValues = value.map((T v) { return v is double ? debugFormatDouble : v.toString(); });
```
Then you could reuse the existing logic for formatting the output in both the line break and regular cases. Someone might add an additional customization for formatting iterables in the future as the current options aren't always ideal so it would be nice to keep one copy of that code.

Reply:
Triggering the transform on whether `T== double` would also be fine with me. E.g

Iterable<Object> formattedValue = T == double ? value.map(debugFormatDouble) : value;

Reply:
Thank you for suggestion. I had tried value.map(debugFormatDouble) but it complained about closure, somehow T == double type check didn't flow through.

Very hard to pass flutter lint rules on this one since v as double is not allowed and T == double keeps v's type as T so you can pass without casting. So (T == double && v is double) was only way to unify since "v is double" won't work on web platform since int and double use num.

I updated the code to a simplified version.